aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -544,20 +543,18 @@ bool canSliceViewTensor(const Tensor& src, MPSShape *mpsShape) {
   }
 
   int64_t sliceOffset = src.storage_offset() / view_numel;
-  // There are cases where both dimensions of a view can shrink
-  // E.g: x = torch.randn((3,6))[1, 1:3]
-  int64_t nextSliceOffset = 0;
-  bool sliceNextDim = (firstDimToSlice < (src_base_shape.size() - 1)) &&
-                      (src_view_shape[firstDimToSlice + 1] != src_base_shape[firstDimToSlice + 1]);
-
   [srcTensorNDArrayDesc sliceDimension:src_ndim_base - 1 - firstDimToSlice withSubrange:{static_cast<NSUInteger>(sliceOffset), static_cast<NSUInteger>(src.sizes()[firstDimToSlice])}];
-  if (sliceNextDim) {
-    if (firstDimToSlice + 1 == src_base_shape.size() - 1) {
-      nextSliceOffset = src.storage_offset() % src_base_shape[src_base_shape.size() - 1];
-    } else {
-      nextSliceOffset = (src.storage_offset() % view_numel) / (view_numel / src_base_shape[firstDimToSlice + 1]);
+
+  // Slice any remaining dimensions
+  for (const auto crtSliceOffset: c10::irange(firstDimToSlice + 1, src_base_shape.size())) {
+    if (src_view_shape[crtSliceOffset] != src_base_shape[crtSliceOffset]) {
+      if (crtSliceOffset == src_base_shape.size() - 1) {
+        sliceOffset = src.storage_offset() % src_base_shape[src_base_shape.size() - 1];
+      } else {
+        sliceOffset = (src.storage_offset() % view_numel) / (view_numel / src_base_shape[crtSliceOffset]);
+      }
+      [srcTensorNDArrayDesc sliceDimension:src_ndim_base - 1 - crtSliceOffset withSubrange:{static_cast<NSUInteger>(sliceOffset), static_cast<NSUInteger>(src.sizes()[crtSliceOffset])}];
```

Comment:
This line is too long. Can you break the withSubRange to next line?

Reply:
Thanks. Updated